envoy/npds.proto

```diff
@@ -0,0 +1,114 @@
+syntax = "proto3";
+
+package pb.cilium;
+// package envoy.api.v2;
+
+import "api/address.proto";
+import "api/discovery.proto";
+import "api/rds.proto";
+
+import "google/api/annotations.proto";
+
+import "validate/validate.proto";
+
+// [#protodoc-title: Network policy management and NPDS]
+
+// Each resource name is a network policy identifier in decimal, e.g. `123`.
+service NetworkPolicyDiscoveryService {
+  rpc StreamNetworkPolicies(stream envoy.api.v2.DiscoveryRequest)
+      returns (stream envoy.api.v2.DiscoveryResponse) {
+  }
+
+  rpc FetchNetworkPolicies(envoy.api.v2.DiscoveryRequest) returns (envoy.api.v2.DiscoveryResponse) {
+    option (google.api.http) = {
+      post: "/v2/discovery:network_policies"
+      body: "*"
+    };
+  }
+}
+
+// A network policy to be enforced on the network flows to/from associated hosts.
+message NetworkPolicy {
+  // The unique identifier of this network policy.
+  // Required.
+  uint64 policy = 1;
+
+  // The part of the policy to be enforced at ingress.
+  // Optional.
+  DirectionNetworkPolicy ingress = 2;
```

Comment:
I found this a little confusing at first, but all OK. This implies that the filter enforcing this policy needs to know if it is part of the ingress or egress pipeline, so the policy enforcement filter needs to be configured accordingly so that it can pick the right policy to enforce.